src/platform/forms-system/src/js/validation.js

```diff
@@ -274,13 +274,9 @@ export function isValidForm(form, pageList) {
             // want to filter the schemas if they can be different. This ensures
             // the data still matches its corresponding schema if we filtered
             // out some data with `itemFilter`.
-            schema.properties[arrayPath] = _.set(
-              'items',
-              schema.properties[arrayPath].items.filter(
-                (item, index) => itemsToKeep[index],
-              ),
-              formData,
-            );
+            schema.properties[arrayPath].items = schema.properties[
+              arrayPath
+            ].items.filter((item, index) => itemsToKeep[index]);
```

Comment:
Since we're using `lodash/fp` here, it looks like we _were_ setting `formData.items` to be the filtered items array. Why don't we need to do that anymore?

Reply:
The update to the form data is still being done at line 261. This part of the code was incorrectly adding the `formData` object to `schema.properties.newDisabilitiesFollowUp`, removing the `type` and `additionalItems`. I don't think the `formData` had anything to do with this `schema`, we needed the `itemFilter` to remove "PTSD" entries since they are handled differently.

Reply:
I would like Chris V to take a look at this, but it is still a pretty significant bug preventing 526 form submission for Veterans, and he won't be back for quite some time.

Reply:
I think this change is good. I just tested the form using the [`upload-781-781a-8940-test.json`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/disability-benefits/all-claims/tests/fixtures/data/upload-781-781a-8940-test.json) data set along with an additional new disability and the `schema` item size changed appropriately upon submission.

We don't want to actually modify the schema, just change the schema item size (by one) based on if "PTSD" is included as one of the new disabilities.

Reply:
I know this passes all unit tests, but it seems like since this bug happened we are either getting a false positive, or we are missing some important coverage. Would it be possible to add a test for this in order to act as a bit of documentation?

Reply:
Done